--- bzip2-1.0.6/Makefile-libbz2_so	2010-09-10 16:07:52 -0700
+++ bzip2-1.0.6-patched/Makefile-libbz2_so	2018-03-20 08:38:49 -0800
@@ -36,7 +36,7 @@ OBJS= blocksort.o  \
 
 all: $(OBJS)
 	$(CC) -shared -Wl,-soname -Wl,libbz2.so.1.0 -o libbz2.so.1.0.6 $(OBJS)
-	$(CC) $(CFLAGS) -o bzip2-shared bzip2.c libbz2.so.1.0.6
+	$(CC) $(CFLAGS) $(LDFLAGS) -o bzip2-shared bzip2.c libbz2.so.1.0.6
 	rm -f libbz2.so.1.0
 	ln -s libbz2.so.1.0.6 libbz2.so.1.0
 
