diff -purN cmake-3.7.2/CMakeLists.txt cmake-3.7.2-patched/CMakeLists.txt
--- cmake-3.7.2/CMakeLists.txt	2017-01-13 06:05:40.000000000 -0800
+++ cmake-3.7.2-patched/CMakeLists.txt	2018-03-08 15:38:40.081157976 -0800
@@ -463,7 +463,7 @@ macro (CMAKE_BUILD_UTILITIES)
   #---------------------------------------------------------------------
   # Build libuv library.
   if(NOT DEFINED CMAKE_USE_LIBUV)
-    set(CMAKE_USE_LIBUV 1)
+    set(CMAKE_USE_LIBUV 0)
     if(APPLE)
       include(CheckCSourceCompiles)
       check_c_source_compiles("
diff -purN cmake-3.7.2/Modules/GNUInstallDirs.cmake cmake-3.7.2-patched/Modules/GNUInstallDirs.cmake
--- cmake-3.7.2/Modules/GNUInstallDirs.cmake	2017-01-13 06:05:41.000000000 -0800
+++ cmake-3.7.2-patched/Modules/GNUInstallDirs.cmake	2018-03-08 15:39:11.148796836 -0800
@@ -210,9 +210,9 @@ if(NOT DEFINED CMAKE_INSTALL_LIBDIR OR (
           "Please enable at least one language before including GNUInstallDirs.")
       else()
         if("${CMAKE_SIZEOF_VOID_P}" EQUAL "8")
-          set(_LIBDIR_DEFAULT "lib64")
+          set(_LIBDIR_DEFAULT "lib")
           if(DEFINED _GNUInstallDirs_LAST_CMAKE_INSTALL_PREFIX)
-            set(__LAST_LIBDIR_DEFAULT "lib64")
+            set(__LAST_LIBDIR_DEFAULT "lib")
           endif()
         endif()
       endif()
