diff -purN coreutils-8.29/src/stty.c coreutils-8.29-patched/src/stty.c
--- coreutils-8.29/src/stty.c	2017-11-24 13:40:53.000000000 -0800
+++ coreutils-8.29-patched/src/stty.c	2018-03-12 22:32:39.000000000 -0700
@@ -347,11 +347,13 @@ static struct mode_info const mode_info[
 #ifdef FLUSHO
   {"flusho", local, SANE_UNSET | REV, FLUSHO, 0},
 #endif
+#ifndef __osf__
 #if defined TIOCEXT
   {"extproc", local, SANE_UNSET | REV | NO_SETATTR, EXTPROC, 0},
 #elif defined EXTPROC
   {"extproc", local, SANE_UNSET | REV, EXTPROC, 0},
 #endif
+#endif
 
   {"evenp", combination, REV | OMIT, 0, 0},
   {"parity", combination, REV | OMIT, 0, 0},
