--- make-4.2-orig/configure	2016-05-22 06:26:58 -0700
+++ make-4.2-new/configure	2018-02-23 12:18:47 -0800
@@ -11382,7 +11382,7 @@ else
 $as_echo "no" >&6; }
      { $as_echo "$as_me:${as_lineno-$LINENO}: checking If the linker accepts -rdynamic" >&5
 $as_echo_n "checking If the linker accepts -rdynamic... " >&6; }
-     LDFLAGS="$old_LDFLAGS -rdynamic"
+     LDFLAGS="$old_LDFLAGS -Wl,rdynamic"
      cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 int main(){}
@@ -11390,7 +11390,7 @@ _ACEOF
 if ac_fn_c_try_link "$LINENO"; then :
   { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 $as_echo "yes" >&6; }
-        AM_LDFLAGS=-rdynamic
+        AM_LDFLAGS=-Wl,rdynamic
 
 else
   { $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
