diff -purN openssl-1.0.2n/Configure openssl-1.0.2n-patched/Configure
--- openssl-1.0.2n/Configure	2017-12-07 05:16:38.000000000 -0800
+++ openssl-1.0.2n-patched/Configure	2018-03-08 11:51:50.624262980 -0800
@@ -365,7 +365,7 @@ my %table=(
 #
 # For gcc, the following gave a %50 speedup on a 164 over the 'DES_INT' version
 #
-"osf1-alpha-gcc", "gcc:-O3::(unknown):::SIXTY_FOUR_BIT_LONG RC4_CHUNK DES_UNROLL DES_RISC1:${alpha_asm}:dlfcn:alpha-osf1-shared:::.so",
+"osf1-alpha-gcc", "gcc:-O3::-pthread:::SIXTY_FOUR_BIT_LONG RC4_CHUNK DES_UNROLL DES_RISC1:${alpha_asm}:dlfcn:alpha-osf1-shared:::.so",
 "osf1-alpha-cc",  "cc:-std1 -tune host -O4 -readonly_strings::(unknown):::SIXTY_FOUR_BIT_LONG RC4_CHUNK:${alpha_asm}:dlfcn:alpha-osf1-shared:::.so",
 "tru64-alpha-cc", "cc:-std1 -tune host -fast -readonly_strings::-pthread:::SIXTY_FOUR_BIT_LONG RC4_CHUNK:${alpha_asm}:dlfcn:alpha-osf1-shared::-msym:.so",
 
diff -purN openssl-1.0.2n/Makefile.shared openssl-1.0.2n-patched/Makefile.shared
--- openssl-1.0.2n/Makefile.shared	2017-12-07 05:16:38.000000000 -0800
+++ openssl-1.0.2n-patched/Makefile.shared	2018-03-08 12:29:32.324938034 -0800
@@ -324,11 +324,11 @@ link_o.alpha-osf1:
 			SHLIB_HIST="$(LIBVERSION)"; \
 		fi; \
 		SHLIB_SOVER=; \
-		ALLSYMSFLAGS='-all'; \
-		NOALLSYMSFLAGS='-none'; \
-		SHAREDFLAGS="$(CFLAGS) $(SHARED_LDFLAGS) -shared -Wl,-B,symbolic"; \
+		ALLSYMSFLAGS='-Wl,-all'; \
+		NOALLSYMSFLAGS='-Wl,-none'; \
+		SHAREDFLAGS="$(CFLAGS) $(SHARED_LDFLAGS) -shared"; \
 		if [ -n "$$SHLIB_HIST" ]; then \
-			SHAREDFLAGS="$$SHAREDFLAGS -set_version $$SHLIB_HIST"; \
+			SHAREDFLAGS="$$SHAREDFLAGS -Wl,-set_version,$$SHLIB_HIST"; \
 		fi; \
 	fi; \
 	$(LINK_SO_O)
@@ -345,11 +345,11 @@ link_a.alpha-osf1:
 			SHLIB_HIST="$(LIBVERSION)"; \
 		fi; \
 		SHLIB_SOVER=; \
-		ALLSYMSFLAGS='-all'; \
-		NOALLSYMSFLAGS='-none'; \
-		SHAREDFLAGS="$(CFLAGS) $(SHARED_LDFLAGS) -shared -Wl,-B,symbolic"; \
+		ALLSYMSFLAGS='-Wl,-all'; \
+		NOALLSYMSFLAGS='-Wl,-none'; \
+		SHAREDFLAGS="$(CFLAGS) $(SHARED_LDFLAGS) -shared"; \
 		if [ -n "$$SHLIB_HIST" ]; then \
-			SHAREDFLAGS="$$SHAREDFLAGS -set_version $$SHLIB_HIST"; \
+			SHAREDFLAGS="$$SHAREDFLAGS -Wl,-set_version,$$SHLIB_HIST"; \
 		fi; \
 	fi; \
 	$(LINK_SO_A)
