--- ppl-0.12.1-orig/src/Watchdog.cc	2012-04-16 05:12:30 -0700
+++ ppl-0.12.1/src/Watchdog.cc	2018-02-24 18:49:26 -0800
@@ -98,7 +98,7 @@ my_getitimer(int which, struct itimerval
 void
 my_setitimer(int which,
 	     const struct itimerval* value, struct itimerval* old_value) {
-  if (setitimer(which, value, old_value) != 0)
+  if (setitimer(which, (struct itimerval *)value, old_value) != 0)
     throw_syscall_error("setitimer");
 }
 
